app/src/main/java/io/homeassistant/companion/android/settings/SettingsFragment.kt

```diff
@@ -100,6 +109,15 @@ class SettingsFragment : PreferenceFragmentCompat(), SettingsView {
             }
         }
 
+        removeSystemFromThemesIfNeeded()
+
+        findPreference<Preference>("themes")?.let {
+            it.summary = (it as ListPreference).entry
+        }
```

Comment:
Can we just use `app:useSimpleSummaryProvider="true"` in the preferences.xml?

Reply:
Ah perfect. I've tried to use this, but android studio didn't suggested me this setting in the autocompletion. So i thought, this property is not supported by ListPreference. But according to the android docs it is supperted :+1: I will change that.

Reply:
I've changed to `app:useSimpleSummaryProvider`. But i needed also to change to use `onSharedPreferenceChanged `because `OnPreferenceChangeListener `triggers before the the value is written to shared preferences. 
If i use `OnPreferenceChangeListener` to trigger theme change, the activity gets reset and the new value never gets written to share prefrences.